File: app/src/main/java/org/mifos/mobile/ui/adapters/BeneficiarySpinnerAdapter.java
package org.mifos.mobile.ui.adapters;

/*
 * Created by saksham on 18/June/2018
 */

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import org.mifos.mobile.R;
import org.mifos.mobile.models.beneficiary.BeneficiaryDetail;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import butterknife.BindView;
import butterknife.ButterKnife;

public class BeneficiarySpinnerAdapter extends ArrayAdapter<String> {

    @BindView(R.id.tv_account_number)
    TextView tvAccountNumber;

    @BindView(R.id.tv_beneficiary_name)
    TextView tvBeneficiaryName;

    int resource;
    LayoutInflater layoutInflater;
    Context context;
    List<BeneficiaryDetail> list;

    public BeneficiarySpinnerAdapter(Context context, int resource, List list) {
        super(context, resource, 0, list);
        this.resource = resource;
        this.context = context;
        this.list = list;
        layoutInflater = LayoutInflater.from(context);
    }

    @NonNull
    @Override
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
        return createItemView(position, convertView, parent);
    }

    @Override
    public View getDropDownView(int position, @Nullable View convertView,
            @NonNull ViewGroup parent) {
        return createItemView(position, convertView, parent);
    }

    public View createItemView(int position, View convertView, ViewGroup parent) {
        View view = layoutInflater.inflate(resource, parent, false);
        ButterKnife.bind(this, view);

        tvAccountNumber.setText(list.get(position).getAccountNumber());
        tvBeneficiaryName.setText(list.get(position).getBeneficiaryName());
        return view;
    }

    @Nullable
    @Override
    public String getItem(int position) {
        return list.get(position).getAccountNumber();
    }
}

File: app/src/main/java/org/mifos/mobile/ui/adapters/BeneficiarySpinnerAdapter.kt
package org.mifos.mobile.ui.adapters

/*
* Created by saksham on 18/June/2018
*/
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import org.mifos.mobile.core.model.entity.beneficiary.BeneficiaryDetail
import org.mifos.mobile.databinding.BeneficiarySpinnerLayoutBinding

class BeneficiarySpinnerAdapter(
    context: Context,
    private var resource: Int,
    private var list: MutableList<BeneficiaryDetail?>,
) : ArrayAdapter<String?>(context, resource, 0, list as List<String?>) {

    private var layoutInflater: LayoutInflater = LayoutInflater.from(context)
    private lateinit var binding: BeneficiarySpinnerLayoutBinding

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        return createItemView(position, convertView, parent)
    }

    override fun getDropDownView(position: Int, convertView: View?, parent: ViewGroup): View {
        return createItemView(position, convertView, parent)
    }

    private fun createItemView(position: Int, convertView: View?, parent: ViewGroup): View {
        val binding = BeneficiarySpinnerLayoutBinding.inflate(layoutInflater, parent, false)
        val view = binding.root
        this.binding = binding

        binding.tvAccountNumber.text = list[position]?.accountNumber
        binding.tvBeneficiaryName.text = list[position]?.beneficiaryName

        return view
    }

    override fun getItem(position: Int): String? {
        return list[position]?.accountNumber
    }
}
